packages/cicero-server/app.js

```diff
@@ -35,47 +48,74 @@ app.use(bodyParser.json());
 app.set('port', PORT);
 
 /**
- * Handle POST requests to /trigger/:template
- * The clause is created using the template and the data.
+ * Handle POST requests to /trigger/:type/:path
+ * The clause is created using the template and the data or the contract instance is created.
  *
- * Template
+ * Type
  * ----------
- * The template parameter is the name of a directory under CICERO_DIR that contains
- * the template to use.
+ * The type parameter is the type that specifies what is to be triggered, a clause or contract.
+ *
+ * Instance Name
+ * ----------
+ * The instanceName parameter is the name of a directory under CICERO_DIR or CICERO_CONTRACTS that contains
+ * the template/contract to use.
  *
  * Request
  * ----------
  * The POST body contains three properties:
  *  - request
+ *  - partyName
  *  - data
  *  - state (optional - for stateless execution)
+ *  - output (optional - for saving updated contract instance in a new slc file under CICERO_CONTRACTS
  *
  * Response
  * ----------
  * JSON formated response object
  *
  */
-app.post('/trigger/:template', async function (req, httpResponse, next) {
+app.post('/trigger/:type/:instanceName', async function (req, httpResponse, next) {
     try {
-        const clause = await initTemplateInstance(req);
+        const instance = req.params.type === 'clause' ? await initTemplateInstance(req) : req.params.type === 'contract' ? await loadInstance(req) : null;
```

Comment:
Why does the caller have to specify the type? We know the type from the template....

Reply:
TBD.